package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1231,22 +1266,78 @@ def print(self, verbose, *args):
+    def _correct_intermittency(self, intermittency, selected_ids):
         """
-        Computes and returns the survival probability timeseries
+        Pre-process Consecutive Intermittency
+        :param intermittency: the max gap allowed and to be corrected
+        :param selected_ids: modifies the selecteded IDs in place by adding atoms which left for <= :param intermittency
+        """
+        self.print('Correcting the selected IDs for intermittancy (gaps). ')
+        if intermittency == 0:
+            return
+
+        # If the atom is absent for a number of frames equal or smaller
+        # than the parameter intermittency, then correct the data and remove the absence.
+        # ie 7,A,A,7 with intermittency=2 will be replaced by 7,7,7,7, where A=absence
+        for i, ids in enumerate(selected_ids):
+            # initially update each frame as seen 0 ago (now)
+            seen_frames_ago = {i: 0 for i in ids}
+            for j in range(1, intermittency + 2):
+                for atomid in seen_frames_ago.keys():
+                    # run out of atoms
+                    if i + j >= len(selected_ids):
+                        continue
+
+                    # if the atom is absent now
+                    if not atomid in selected_ids[i + j]:
+                        # increase its absence counter
+                        seen_frames_ago[atomid] += 1
+                        continue
+
+                    # the atom is found
+                    # it was present in the last frame
+                    if seen_frames_ago[atomid] == 0:
+                        continue
+
+                    # we are passed the gap
+                    if seen_frames_ago[atomid] > intermittency:
+                        continue
+
+                    # the atom was absent but returned on time (<= intermittency)
+                    # add it to the frames where it was absent.
+                    # Introduce the corrections.
+                    for k in range(seen_frames_ago[atomid], 0, -1):
+                        selected_ids[i + j - k].add(atomid)
+
+                    seen_frames_ago[atomid] = 0
+
+
+    def run(self, tau_max=20, start=0, stop=None, step=1, residues=False, intermittency=0, verbose=False):
+        """
+        Computes and returns the Survival Probability (SP) timeseries
 
         Parameters
         ----------
-        start : int
+        start : int, optional
             Zero-based index of the first frame to be analysed
-        stop : int
+        stop : int, optional
             Zero-based index of the last frame to be analysed (inclusive)
-        step : int
-            Jump every `step`'th frame
-        tau_max : int
+        step : int, optional
+            Jump every `step`'th frame. This is compatible but independant of the taus used, and it is good to consider
+            using the  step as tau_max to remove the overlap. Note that step and taus are consistent with intermittency.
```

Comment:
put all references parameters in single backticks such as
```reST
Jump every `step` -th frame. This is compatible but independant of the taus used, and it is good to consider
using the  `step` as `tau_max` to remove the overlap.
```
Note that you need a space after the backticks.

Build the docs locally and check that your docs look ok, see https://github.com/MDAnalysis/mdanalysis/wiki/WritingDocumentation#generating-the-docs